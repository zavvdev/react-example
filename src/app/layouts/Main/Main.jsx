import { childrenPropType } from "app/propTypes";
import { useMainStyles } from "app/layouts/Main/Main.styles";
import { Containers } from "app/containers";

export function Main({ children }) {
  const classes = useMainStyles();

  return (
    <div className={classes.root}>
      <div className={classes.inner}>
        <Containers.AppHeader />
        <div className={classes.content}>{children}</div>
      </div>
    </div>
  );
}

Main.propTypes = {
  children: childrenPropType.isRequired,
};
