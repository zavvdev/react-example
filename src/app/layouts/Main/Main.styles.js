import { createUseStyles } from "react-jss";

export const useMainStyles = createUseStyles((theme) => ({
  root: {
    boxSizing: "border-box",
    minHeight: "100vh",
    backgroundColor: theme.backgroundColor,
    padding: "0 5%",
  },
  inner: {
    maxWidth: "500px",
    margin: "0 auto",
    paddingBottom: "200px",
  },
  content: {
    paddingTop: "36px",
  },
}));
