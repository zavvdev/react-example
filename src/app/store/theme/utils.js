import { localStorageService } from "app/services/LocalStorageService";
import { IS_DARK_THEME_LOCAL_STORAGE_KEY } from "app/store/theme/config";

export const saveDarkModeState = (darkModeState) => {
  localStorageService.set(IS_DARK_THEME_LOCAL_STORAGE_KEY, `${darkModeState}`);
};

export const getDarkModeState = () => {
  const rawStorageValue = localStorageService.get(
    IS_DARK_THEME_LOCAL_STORAGE_KEY,
  );
  return !!JSON.parse(rawStorageValue);
};
