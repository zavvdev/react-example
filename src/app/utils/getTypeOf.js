export function getTypeOf(object) {
  const objectType = typeof object;
  let returnType = objectType;
  if (objectType === "object") {
    const stringifiedObject = Object.prototype.toString.call(object);
    returnType = stringifiedObject
      .replace(/^\[object (\S+)\]$/, "$1")
      .toLowerCase();
  }
  return returnType;
}
