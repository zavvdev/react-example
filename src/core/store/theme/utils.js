import { LOCAL_STORAGE_KEYS } from "core/config/localStorage";
import { localStorageService } from "core/services/LocalStorageService";

export const saveDarkModeState = (darkModeState) => {
  localStorageService.set(
    LOCAL_STORAGE_KEYS.isThemeDarkMode,
    `${darkModeState}`,
  );
};

export const getDarkModeState = () => {
  const rawStorageValue = localStorageService.get(
    LOCAL_STORAGE_KEYS.isThemeDarkMode,
  );
  return !!JSON.parse(rawStorageValue);
};
