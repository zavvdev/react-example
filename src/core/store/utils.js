import { all, call, spawn } from "redux-saga/effects";
import { errorTrackingService } from "core/services/ErrorTrackingService";

export const combineMiddleware = (middlewareList) => function* combiner() {
  const getRootGenerator = (middleware) => function* rootGenerator() {
    while (true) {
      try {
        yield call(middleware);
        break;
      } catch (e) {
        errorTrackingService.reportError(e);
      }
    }
  };

  yield all(middlewareList.map((middleware) => {
    return spawn(getRootGenerator(middleware));
  }));
};
