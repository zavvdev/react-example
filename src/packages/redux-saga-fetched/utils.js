import { ACTION_TYPES, DATA_STATUS_TYPES } from "packages/redux-saga-fetched/config";

export const createActionType = ({ createdKey, actionTypePattern }) => {
  return `${createdKey}_${actionTypePattern}`;
};

export const createKey = (key) => key.join("_");

export const createActionTypePatterns = (domain) => ({
  request: `${domain}@${ACTION_TYPES.request}`,
  success: `${domain}@${ACTION_TYPES.success}`,
  failure: `${domain}@${ACTION_TYPES.failure}`,
  invalidate: `${domain}@${ACTION_TYPES.invalidate}`,
  reset: `${domain}@${ACTION_TYPES.reset}`,
});

export const createState = ({
  isLoading,
  isFetching,
  isLoaded,
  isError,
  isValid,
  status,
  data,
}) => ({
  isLoading,
  isFetching,
  isLoaded,
  isError,
  isValid,
  status,
  data,
});

export const createResetState = () => createState({
  isLoading: false,
  isFetching: false,
  isLoaded: false,
  isError: false,
  isValid: false,
  status: DATA_STATUS_TYPES.reset,
  data: null,
});

export const createRequestState = ({ state, payload }) => {
  const isLoading = !state[payload.createdKey]?.data;
  const isFetching = !!state[payload.createdKey]?.data;
  return createState({
    isLoading,
    isFetching,
    isLoaded: false,
    isError: false,
    isValid: false,
    status: isLoading ? DATA_STATUS_TYPES.loading : DATA_STATUS_TYPES.fetching,
    data: state[payload.createdKey]?.data || null,
  });
};

export const createSuccessState = ({ payload }) => createState({
  isLoading: false,
  isFetching: false,
  isLoaded: true,
  isError: false,
  isValid: true,
  status: DATA_STATUS_TYPES.loaded,
  data: payload.data,
});

export const createFailureState = ({ state, payload }) => createState({
  isLoading: false,
  isFetching: false,
  isLoaded: false,
  isError: true,
  isValid: false,
  status: DATA_STATUS_TYPES.error,
  data: state[payload.createdKey]?.data || null,
});

export const createInvalidateState = ({ state, payload }) => createState({
  isLoading: state[payload.createdKey]?.isLoading,
  isFetching: state[payload.createdKey]?.isFetching,
  isLoaded: state[payload.createdKey]?.isLoaded,
  isError: state[payload.createdKey]?.isError,
  isValid: false,
  status: state[payload.createdKey]?.status,
  data: state[payload.createdKey]?.data,
});
