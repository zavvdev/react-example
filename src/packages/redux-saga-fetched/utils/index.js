import { ACTION_TYPES, EFFECT_TYPES } from "packages/redux-saga-fetched/config";

export const createActionType = ({ createdKey, actionTypePattern }) => {
  return `${createdKey}_${actionTypePattern}`;
};

export const createKey = (key) => key.join("_");

export const createActionTypePatterns = (domain) => ({
  [EFFECT_TYPES.query]: {
    request: `${domain}@${ACTION_TYPES[EFFECT_TYPES.query].request}`,
    success: `${domain}@${ACTION_TYPES[EFFECT_TYPES.query].success}`,
    failure: `${domain}@${ACTION_TYPES[EFFECT_TYPES.query].failure}`,
    invalidate: `${domain}@${ACTION_TYPES[EFFECT_TYPES.query].invalidate}`,
    reset: `${domain}@${ACTION_TYPES[EFFECT_TYPES.query].reset}`,
  },
  [EFFECT_TYPES.mutation]: {
    request: `${domain}@${ACTION_TYPES[EFFECT_TYPES.mutation].request}`,
    success: `${domain}@${ACTION_TYPES[EFFECT_TYPES.mutation].success}`,
    failure: `${domain}@${ACTION_TYPES[EFFECT_TYPES.mutation].failure}`,
    reset: `${domain}@${ACTION_TYPES[EFFECT_TYPES.mutation].reset}`,
  },
});
