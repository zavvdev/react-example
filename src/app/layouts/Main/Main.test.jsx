import { screen } from "@testing-library/react";
import { testUtils } from "app/tests/utils";
import { Main } from "app/layouts/Main/Main";

const childrenMock = "childrenMock";

describe("Main layout", () => {
  test("should render Main layout (snapshot)", () => {
    const result = testUtils.render(
      <Main>
        <div>{childrenMock}</div>
      </Main>,
    );
    expect(result.asFragment()).toMatchSnapshot();
  });

  test("should have a children", () => {
    testUtils.render(
      <Main>
        <div>{childrenMock}</div>
      </Main>,
    );
    expect(screen.getByText(childrenMock)).toBeInTheDocument();
  });
});
