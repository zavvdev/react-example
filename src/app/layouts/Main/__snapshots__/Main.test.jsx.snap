// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main layout should render Main layout (snapshot) 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="inner"
    >
      <div
        class="root"
      >
        <div
          class="side"
        >
          <button
            class="root"
            type="button"
          >
            <div
              class="switcher"
            >
              themeSwitch.light
            </div>
          </button>
        </div>
        <a
          class="logo"
          href="/"
        >
          <span
            class="root"
          >
            appName
          </span>
        </a>
        <div
          class="side"
        >
          <div
            class="root"
          >
            <a
              class="cart"
              href="/cart"
            >
              appHeader.cart
            </a>
          </div>
        </div>
      </div>
      <div
        class="content"
      >
        <div>
          childrenMock
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
